### PR TITLE
make vertex and index buffers resizable

### PR DESCRIPTION
**Pull request by ansis**
This isn't used yet but is one small step towards fast updates to layout properties. (I'm pulling the out of a old, unfinished, stale branch).

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [n/a] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [n/a] document any changes to public APIs
 - [n/a] post benchmark scores
 - [x] manually test the debug page

### CODE REVIEW

**Comment by mourner** [comment]
Given that this is a pretty small change, should we include it when we actually use it? Or do you anticipate taking advantage of this relatively soon?

File: src/gl/index_buffer.js [hunk 1]
```diff
@@ -35,12 +39,17 @@ class IndexBuffer {
...
         // The right VAO will get this buffer re-bound later in VertexArrayObject#bind
         // See https://github.com/mapbox/mapbox-gl-js/issues/5620
         this.context.unbindVAO();
         this.bind();
-        gl.bufferSubData(gl.ELEMENT_ARRAY_BUFFER, 0, array.arrayBuffer);
+        if (this.length === array.length) {
+            assert(this.dynamicDraw || this.resizable);
+            gl.bufferSubData(gl.ELEMENT_ARRAY_BUFFER, 0, array.arrayBuffer);
+        } else {
+            assert(this.resizable);
```
Comment (arindam1993) [inline src/gl/index_buffer.js]: Is the resizable boolean really necessary, I assume it is so we can be more explicit about when we want to pay the performance cost of doing a full bufferData.
But could we also do that by adding a warning thats stripped away in production?

Comment (kkaefer) [reply src/gl/index_buffer.js]: Is there any advantage to having a buffer non-resizable? Why aren't all buffers resizable by default? We don't necessarily need to resize them.

**Comment by ansis** [comment]
Closing for now. I'll try to merge it as part of a larger pr.

**Pull request closed without merging**